paper/paper.md

```diff
@@ -84,22 +84,22 @@ $$
+Essentially, in each trace the Okada filter replaces the point $x_{t}$ with the average of adjacent values when the product of the differences in adjacent values is greater than zero. One useful characteristic of this smoothing algorithm is that it does not move the start position of events like other algorithms do [@Okada2016]
 
 ## Nonnegative Deconvolution
-`pyNeuroTrace` also has an implementation of nonnegative deconvolution (NND) to be applied to photocurrents in order to reduce noise in raw time series recordings [@Podgorski2012]. Another application of this algorithm is its use in event detection in neuronal activity traces, which, due to biosensor kinetics, follow similar decays as photocurrents from detectors [@Podgorski2012]. This can be particularly useful for finding smaller magnitude events in fluorescent imaging that are often obfuscated by machine noise [@Podgorski2012].
+When the kinetics of biosensors are known, nonnegative deconvolution (NND) can be useful to deconvolve raw or $\Delta F/F$ traces back into the most likely discrete triggers of increases in the sensors. `pyNeuroTrace` includes an implementation of an algorithm that can efficiently compute NND in linear time [@Podgorski2012]. This can also be useful for denoising, as well particularly good at finding smaller magnitude events in fluorescent imaging that are often obfuscated by machine noise [@Podgorski2012].
 
 # Event Detection
-The event detection module uses several strategies to identify neuronal activity events in time series datasets. These methodologies have been previously discussed and compared by Sakaki *et al.* [@Sakaki2018]. These include two generalizable methods and one that requires prior knowledge of recorded event shape. The generalizable methods include filtering the signal through an exponentially weighted moving average (ewma) or cumulative sum of movement above the mean (cusum). The final filter is a matched filter that finds the probability of the trace matching a previously defined shape, such as one described by an exponential rise and decay of calcium signal generated by a GECI.
+The event detection module uses several strategies to identify neuronal activity events in time series datasets. These methodologies have been previously discussed and compared by Sakaki *et al.* [@Sakaki2018]. These include two generalizable methods and one that requires prior knowledge of recorded event shape. The generalizable methods include filtering the signal through an exponentially weighted moving average (ewma) or cumulative sum of movement above the mean (cusum). The final filter is a matched filter that finds the probability of the trace matching a previously defined shape, such as one described by an exponential rise and decay of calcium signal generated by a GECI. Each algorithm was chosen as they are is computationally efficient, so are able to be run in real-time during an experiment if required, or process very large time series.
```

Comment:
Computationally efficient algorithms were selected, allowing for the possibility of applying them in real-time during experiments.